libs/api/utils/src/lib/api-utils.ts

```diff
@@ -1,21 +1,53 @@
 import { Prisma, prisma } from '@nmit-coursition/db'
-import { isDateBeforeNow } from '@nmit-coursition/utils'
-import { parseApiKey } from '../api'
+import { generateRandomIdentifier, isDateBeforeNow } from '@nmit-coursition/utils'
+import { Elysia } from 'elysia'
+import { formatApiErrorResponse, parseApiKey } from '../api'
 import type { ApiErrorCode } from '../errorList'
-import type { ApiKeyReportUsageRequest, ApiUsageReport, ApiUsageRequest } from '../typescript'
+import { ERROR_LIST } from '../errorList'
+import { errorResponseModel } from '../model'
+import type { ApiKeyReportUsageRequest, ApiUsageReport, ApiUsageRequest, ExtendedRequest } from '../typescript'
 
 let API_KEY_TO_ID_CACHE: { [key: string]: bigint } = {}
 
-let API_KEY_CONTEXT: string | undefined
+export const apiCommonGuard = new Elysia().guard({
+  as: 'global',
+  response: {
+    401: errorResponseModel,
+    404: errorResponseModel,
+    429: errorResponseModel,
+    500: errorResponseModel,
+  },
+  detail: {
+    tags: ['v1'],
```

Comment:
**suggestion:** As this is guard, it shouldn't have tag of "v1".

Reply:
It has been copy-pasted from original endpoint implementation.